lib/puppet/functions/seeded_rand.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+# @summary
+#   Generates a random whole number greater than or equal to 0 and less than max, using the value of seed for repeatable randomness.
+Puppet::Functions.create_function(:seeded_rand) do
```

Comment:
nit: i think it would be nice to create this as a namespaced function with a legacy deprecated shim e.g.
https://github.com/puppetlabs/puppetlabs-stdlib/pull/1341/files